### PR TITLE
MMCA-4917 | Implement CSV hyperlink to download on cash account transaction - V2

### PR DESCRIPTION
**Pull request by Neozxc**
Task:
- Query SDES service for cash account csv files in cash account frontend
- Create a new view for cash account statements
- Map the requested transactions link to the new page
- Add unit tests

### CODE REVIEW

File: app/helpers/Formatters.scala [hunk 1]
```diff
@@ -35,6 +35,13 @@ trait DateFormatters {
     s"${dateAsMonth(date)}-${date.getYear}"
 
   def dateAsDay(date: LocalDate): String = DateTimeFormatter.ofPattern("d").format(date)
+
+  def dateAsMonthToMonth(periodStartMonth: Int, periodEndMonth: Int)(implicit messages: Messages): String = {
```
Comment (AdamCoops) [inline app/helpers/Formatters.scala:39]: Can we write a test for this if there isn't one already?

File: conf/messages.cy [hunk 2]
```diff
@@ -199,6 +200,17 @@ cf.common.missing-documents-guidance.text2=Cynhyrchir {0} dim ond ar gyfer cyfno
 cf.import-vat.requested.title=Tystysgrifau TAW mewnforio (C79) y gofynnwyd amdanynt
 cf.import-vat.requested.available.text=Mae tystysgrifau y gofynnwyd amdanynt ar gael i’w gweld am 10 diwrnod. Gallwn ddarparu’r rhain ar ffurf PDF yn unig.
 
+# Cash Requested Statements View
+cf.cash-statement-requested-account-title=
```
Comment (AdamCoops) [inline conf/messages.cy:204]: Does this need a value? Can we remove it if it's not referenced?

**Comment by tyagikumarvineet** [comment]
Could you please address below scalastyle warnings?

[warn] /Users/vineet.tyagi/hmrc/codebase/customs-historic-statement-frontend/test/views/HistoricDateRequestPageViewSpec.scala:21: File line length exceeds 160 characters
[warn] /Users/vineet.tyagi/hmrc/codebase/customs-historic-statement-frontend/test/views/CashStatementViewSpec.scala:32:0: Whitespace at end of line

File: test/viewmodels/CashStatementViewModelSpec.scala
Comment (tyagikumarvineet) [inline test/viewmodels/CashStatementViewModelSpec.scala:1]: It should be updated, once the relevant ViewModel is refactored. (as part of separate ticket)

File: test/views/components/DownloadLinkSpec.scala [hunk 3]
```diff
@@ -17,38 +17,58 @@
 package views.components
 
 import helpers.Formatters
-import models.FileFormat.Pdf
-import models.{C79Certificate, FileFormat, VatCertificateFile, VatCertificateFileMetadata}
+import models.FileFormat.{Csv, Pdf}
+import models._
```
Comment (tyagikumarvineet) [inline test/views/components/DownloadLinkSpec.scala:21]: Its better to revert back to only importing specific models that we need, as its good practice to provide the specific models (that are being used) and also it increases the compile time.

File: app/config/FrontendAppConfig.scala [hunk 4]
```diff
@@ -17,7 +17,7 @@
 package config
 
 import com.google.inject.{Inject, Singleton}
-import models.{C79Certificate, DutyDefermentStatement, FileRole, PostponedVATStatement, SecurityStatement, UserAnswers}
+import models._
```
Comment (tyagikumarvineet) [inline app/config/FrontendAppConfig.scala:20]: Could you please revert back as its good practice to specify the specific models (that are being used) and using all the models increase the compile time. We should only use ._ when either we are using most of the models form the folder or all of them. Here we are only using only 6 out of lot

File: app/views/components/download_link_cash_account.scala.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ * limitations under the License.
+ *@
+
+@this()
+
+@(file: Option[CashStatementFile], fileFormat: FileFormat, id: String, period: String)(implicit messages: Messages)
+
+@if(file.isDefined) {
+    <a id="@id" class="file-link govuk-link" href="@{file.get.downloadURL}" download>
+        <span>@fileFormat (@file.get.formattedSize)</span>
```
Comment (wg-hmrc) [inline app/views/components/download_link_cash_account.scala.html:23]: Please apply `aria-hidden="true"` here as well

**Review by tyagikumarvineet — APPROVED** [review]
Rest of the comments would be addressed in a separate ticket. Approving now.

**Pull request merged**